.github/workflows/python-test-uv.yml

```
@@ -29,13 +33,26 @@ jobs:
         run: |
           uv sync --all-extras --dev --group lint
           uv add --dev pytest pytest-cov coverage
+          uv add --dev basedpyright
 
       - name: format check
         run: uv run ruff format --check
 
       - name: lint
         run: uv run ruff check
 
+      - name: typecheck
+        continue-on-error: ${{ inputs.require_typecheck == false }}
+        shell: bash
+        run: |
+          uv run basedpyright || {
+          	exit_code=$?
+          	if [[ $exit_code -eq 1 && "${{ inputs.require_typecheck }}" == "false" ]]; then
+          		echo "::warning title=Typechecking::Errors found"
+          	fi
+          		exit $exit_code
+          	}
```

```suggestion
          uv run basedpyright || {
            exit_code=$?
            if [[ $exit_code -eq 1 && "${{ inputs.require_typecheck }}" == "false" ]]; then
              echo "::warning title=Typechecking::Errors found"
            fi
            exit $exit_code
          }
```
spaces over tabs :)))
but all jokes aside, you have mixed indentation levels and the last command should be aligned with the `fi`. There is no linter to fix this issue, cus this whole thing is a `string` for yaml, so it can't know ur intent here :(

---

fixed bc its cut